src/modules/orders/orders.service.ts

```diff
@@ -0,0 +1,193 @@
+        throw new NotFoundException(Errors.USER_NOT_FOUND);
+      }
+
+      const orders = await this.orderRepository.find({
+        where: { vendorId: vendorId },
+        relations: ['products'],
+        order: { createdAt: 'DESC' },
+      });
+
+      if (!orders.length) {
+        throw new NotFoundException(Errors.ORDERS_NOT_FOUND);
+      }
+
+      return orders.map((order) => new OrderResponseDTO(order));
+    } catch (error) {
+      if (error instanceof NotFoundException) {
+        throw error;
+      }
+      throw new InternalServerErrorException(
+        Errors.FAILED_TO_FETCH_ORDERS_BY_VENDOR,
+      );
+    }
+  }
+
+  async createOrdersForUser(
+    userId: string,
+    shippingPrice: number,
+  ): Promise<void> {
+    try {
+      const user = await this.userRepository.findOne({ where: { id: userId } });
+
+      if (!user) {
+        throw new NotFoundException(Errors.USER_NOT_FOUND);
+      }
+
+      const cartItems = await this.cartRepository.find({
+        where: { userId: userId },
+      });
+
+      if (!cartItems.length) {
+        throw new NotFoundException(Errors.CART_IS_EMPTY);
+      }
+
+      const ordersMap = new Map<string, OrderDTO>();
+
+      for (const cartItem of cartItems) {
+        const vendor = await this.userRepository.findOne({
+          where: { id: cartItem.vendorId },
+        });
+
+        if (!vendor) {
+          throw new NotFoundException(Errors.VENDOR_NOT_FOUND);
+        }
+
+        const productResponseDTO = await this.productRepository.findOne({
+          where: { id: cartItem.productId },
+        });
+
+        if (!productResponseDTO) {
+          throw new NotFoundException(Errors.PRODUCT_NOT_FOUND);
+        }
+
+        const product = {
+          productId: cartItem.productId,
+          productUrl: cartItem.productUrl,
+          name: productResponseDTO.name,
+          size: cartItem.size,
+          color: cartItem.color,
+          duration: cartItem.duration,
+          price: cartItem.price,
+        };
+
+        if (ordersMap.has(cartItem.vendorId)) {
+          ordersMap.get(cartItem.vendorId).products.push(product);
+        } else {
+          ordersMap.set(cartItem.vendorId, {
+            vendorId: vendor.id,
+            vendorName: vendor.name,
+            products: [product],
+          });
+        }
+      }
+
+      const orders = Array.from(ordersMap.values());
+
+      if (!orders.length) {
+        throw new NotFoundException(Errors.ORDERS_CANNOT_BE_GENERATED);
+      }
+
+      const buyerOrder = new BuyerOrder();
+
+      let totalPrice = 0;
+
+      cartItems.forEach((item) => {
+        totalPrice += parseFloat(String(item.price));
+      });
+
+      totalPrice += parseFloat(String(shippingPrice));
+
+      buyerOrder.user = user;
+      buyerOrder.status = Status.NEW_ORDER;
+      buyerOrder.shipping = shippingPrice;
+      buyerOrder.price = totalPrice;
+      buyerOrder.orders = [];
+
+      for (const orderDTO of orders) {
```

Comment:
orderDTO is very bad naming, name should describe what variable save, DTO say nothing

Reply:
fixed